pkg/networkservice/common/mechanisms/recvfd/common.go

```diff
@@ -86,7 +86,7 @@ func recvFDAndSwapInodeToFile(ctx context.Context, fileMap *perConnectionFileMap
 	return err
 }
 
-func swapFileToInode(fileMap *perConnectionFileMap, parameters map[string]string, closeAllFiles bool) error {
+func swapFileToInode(fileMap *perConnectionFileMap, parameters map[string]string) error {
```

Comment:
Have you tested this in integration?  I ask... because I would expect this to cause issues.  The purpose of the recvfd was to receive the file *and hold it open* for the life of the connection.  This is because *other things* like for example memifproxy may need to be using it during the life of the connection.

Now please note, the *and hold it open* part may in fact not be necessary... but you will not be able to tell that from the unit testing in SDK.  Testing in integration or with cmd-forwarder-vpp would give you a sense of whether you can get away with closing these files *after* the processing of a Request message or whether you need to hold them open for the life of the connection.

I'm hoping your are correct here... because I'd like to see this change work as submitted :)

Reply:
@edwarnicke , thanks a lot for your comments!

As far as I understand, if there's a timeout, [all files are also closed](https://github.com/networkservicemesh/sdk/pull/1173/files#diff-8fdf36a9c6dd5a624ddd9ded68242ae81e6f081ca16441e1eca7ce4a6ae4954aL119) as `closeAllFiles` flag is `true` [in Close method](https://github.com/networkservicemesh/sdk/pull/1173/files#diff-17df730ee1c0d28ce8e652baf843f0b8b3af6f77635d07026bef56db3b3f3586L107). The cleanup I added is for the case when there's no `Peer` in context (`cancelContext` passed from `timeoutserver`).

I'm definitely **not** going to commit this unless it's tested in integration (locally as well as on CI - will discuss the details with Denis).

Reply:
@Sol-0 Yeah... and please note: your change may be right.  I *hope* its right.  I was just wracking my brains as to why the closeAll had been there to begin with, and thought of *possible* (not certain) issues.

Reply:
@denis-tingaikin , @edwarnicke integration testing has been completed: https://github.com/networkservicemesh/integration-k8s-kind/actions/runs/1517061295

Local integration testing has also completed successfully.